### ISSUE TITLE
clean install leads to "write error" in dashboard graph

### ISSUE THREAD

**Issue by igarny** [perfsonar/graphs#136]
Hello Michael,

For me a clean install of **perfsonar-centralmanagement** on a CentOS 7.5 system after adding the missing perl library is showing errors in graphs, although the state of matrix is correctly presented. Error is:
```
Error retrieving data
write failed:
```
there was nothing after the double column.

In the apache logs I have **500** error messages:
```
249 - - [16/Nov/2018:09:29:55 +0000] "GET /maddash-webui/details.cgi?uri=/maddash/grids/GEANT+Global+Mesh+-+IPv4+iperf3+-+Throughput/psmp-gn-bw-01-poz-pl-v4.geant.net/psmp-gn-bw-01-lon-uk.geant.net/Throughput HTTP/1.1" 200 4438
249 - - [16/Nov/2018:09:29:56 +0000] "GET /maddash/dashboards HTTP/1.1" 200 1543
249 - - [16/Nov/2018:09:29:57 +0000] "GET /maddash/grids HTTP/1.1" 200 1499
249 - - [16/Nov/2018:09:29:56 +0000] "GET /maddash/grids/GEANT%20Global%20Mesh%20-%20IPv4%20iperf3%20-%20Throughput/psmp-gn-bw-01-poz-pl-v4.geant.net/psmp-gn-bw-01-lon-uk.geant.net/Throughput HTTP/1.1" 200 9972
249 - - [16/Nov/2018:09:29:57 +0000] "GET /maddash/events?gridName=GEANT%20Global%20Mesh%20-%20IPv4%20iperf3%20-%20Throughput&rowName=psmp-gn-bw-01-poz-pl-v4.geant.net&columnName=psmp-gn-bw-01-lon-uk.geant.net&checkName=Throughput HTTP/1.1" 200 13
249 - - [16/Nov/2018:09:29:57 +0000] "GET /perfsonar-graphs/graphWidget.cgi?source=psmp-gn-bw-01-poz-pl-v4.geant.net&dest=psmp-gn-bw-01-lon-uk.geant.net&displaysetdest=&url=https://prod-psma-gn-01-buc-ro.geant.net/esmond/perfsonar/archive/&reverseurl=https://prod-psma-gn-01-buc-ro.geant.net/esmond/perfsonar/archive/&displaysetsrc= HTTP/1.1" 301 536
249 - - [16/Nov/2018:09:29:57 +0000] "GET /perfsonar-graphs/?source=psmp-gn-bw-01-poz-pl-v4.geant.net&dest=psmp-gn-bw-01-lon-uk.geant.net&displaysetdest=&url=https://prod-psma-gn-01-buc-ro.geant.net/esmond/perfsonar/archive/&reverseurl=https://prod-psma-gn-01-buc-ro.geant.net/esmond/perfsonar/archive/&displaysetsrc= HTTP/1.1" 200 783
83 - - [16/Nov/2018:09:29:57 +0000] "POST /esmond/perfsonar/archive/ HTTP/1.1" 201 5438
83 - - [16/Nov/2018:09:29:57 +0000] "PUT /esmond/perfsonar/archive/a427b84ca4314b3dbaf87898f19e0a46/ HTTP/1.1" 201 2
249 - - [16/Nov/2018:09:29:57 +0000] "GET /perfsonar-graphs/cgi-bin/graphData.cgi?action=hosts&src=psmp-gn-bw-01-poz-pl-v4.geant.net&dest=psmp-gn-bw-01-lon-uk.geant.net HTTP/1.1" 200 177
249 - - [16/Nov/2018:09:29:58 +0000] "GET /perfsonar-graphs/cgi-bin/graphData.cgi?action=ma_data&url=https%3A%2F%2Fprod-psma-gn-01-buc-ro.geant.net%2Fesmond%2Fperfsonar%2Farchive%2F%3Fsource%3Dpsmp-gn-bw-01-lon-uk.geant.net%26destination%3Dpsmp-gn-bw-01-poz-pl-v4.geant.net HTTP/1.1" 500 14
249 - - [16/Nov/2018:09:29:57 +0000] "GET /perfsonar-graphs/cgi-bin/graphData.cgi?action=has_traceroute_data&url=https://prod-psma-gn-01-buc-ro.geant.net/esmond/perfsonar/archive/&source=psmp-gn-bw-01-poz-pl-v4.geant.net&dest=psmp-gn-bw-01-lon-uk.geant.net HTTP/1.1" 200 316
249 - - [16/Nov/2018:09:29:58 +0000] "GET /perfsonar-graphs/cgi-bin/graphData.cgi?action=ma_data&url=https%3A%2F%2Fprod-psma-gn-01-buc-ro.geant.net%2Fesmond%2Fperfsonar%2Farchive%2F%3Fsource%3Dpsmp-gn-bw-01-poz-pl-v4.geant.net%26destination%3Dpsmp-gn-bw-01-lon-uk.geant.net HTTP/1.1" 500 14
83 - - [16/Nov/2018:09:30:00 +0000] "POST /esmond/perfsonar/archive/ HTTP/1.1" 201 5441
```

**Comment by igarny** [comment]
![image](https://user-images.githubusercontent.com/11942774/48717019-e16a2400-ec18-11e8-827f-28826e779452.png)

**Comment by igarny** [comment]
these are all the events in the Apache error log:
```
[Thu Nov 15 16:03:05.668355 2018] [:error] [pid 43052] cassandra_db [INFO] Checking/creating column families
[Thu Nov 15 16:03:05.670306 2018] [:error] [pid 43052] cassandra_db [INFO] Schema check done
[Thu Nov 15 16:03:05.670366 2018] [:error] [pid 43052] cassandra_db [DEBUG] Opening ConnectionPool
[Thu Nov 15 16:03:05.678363 2018] [:error] [pid 43052] cassandra_db [INFO] Connected to ['localhost:9160']
[Thu Nov 15 16:03:05.702338 2018] [:error] [pid 43052] cassandra_db [INFO] Checking/creating column families
[Thu Nov 15 16:03:05.702384 2018] [:error] [pid 43052] cassandra_db [INFO] Checking/creating column families
[Thu Nov 15 16:03:05.704431 2018] [:error] [pid 43052] cassandra_db [INFO] Schema check done
[Thu Nov 15 16:03:05.706622 2018] [:error] [pid 43052] cassandra_db [INFO] Schema check done
[Thu Nov 15 16:03:05.706690 2018] [:error] [pid 43052] cassandra_db [DEBUG] Opening ConnectionPool
[Thu Nov 15 16:03:05.706709 2018] [:error] [pid 43052] cassandra_db [DEBUG] Opening ConnectionPool
[Thu Nov 15 16:03:05.715690 2018] [:error] [pid 43052] cassandra_db [INFO] Connected to ['localhost:9160']
[Thu Nov 15 16:03:05.715726 2018] [:error] [pid 43052] cassandra_db [INFO] Connected to ['localhost:9160']
[Fri Nov 16 09:26:15.838426 2018] [mpm_prefork:notice] [pid 43043] AH00170: caught SIGWINCH, shutting down gracefully
[Fri Nov 16 09:26:17.934734 2018] [suexec:notice] [pid 103082] AH01232: suEXEC mechanism enabled (wrapper: /usr/sbin/suexec)
[Fri Nov 16 09:26:17.955096 2018] [so:warn] [pid 103082] AH01574: module ssl_module is already loaded, skipping
[Fri Nov 16 09:26:17.957813 2018] [auth_digest:notice] [pid 103082] AH01757: generating secret for digest authentication ...
[Fri Nov 16 09:26:17.958504 2018] [lbmethod_heartbeat:notice] [pid 103082] AH02282: No slotmem from mod_heartmonitor
[Fri Nov 16 09:26:17.966874 2018] [mpm_prefork:notice] [pid 103082] AH00163: Apache/2.4.6 (CentOS) OpenSSL/1.0.2k-fips mod_wsgi/3.4 Python/2.7.5 configured -- resuming normal operations
[Fri Nov 16 09:26:17.966902 2018] [core:notice] [pid 103082] AH00094: Command line: '/usr/sbin/httpd -D FOREGROUND'
[Fri Nov 16 09:26:21.927464 2018] [:error] [pid 103089] path= ['/usr/lib/esmond/esmond', '/usr/lib/esmond/lib/python2.7', '/usr/lib/esmond/lib/python2.7/site-packages', '/usr/lib/esmond/src/dlnetsnmp/lib', '/usr/lib/esmond', '/usr/lib/esmond/esmond_client', '/usr/lib64/python27.zip', '/usr/lib64/python2.7', '/usr/lib64/python2.7/plat-linux2', '/usr/lib64/python2.7/lib-tk', '/usr/lib64/python2.7/lib-old', '/usr/lib64/python2.7/lib-dynload', '/usr/lib64/python2.7/site-packages', '/usr/lib64/python2.7/site-packages/gtk-2.0', '/usr/lib/python2.7/site-packages']
[Fri Nov 16 09:26:21.928893 2018] [:error] [pid 103084] path= ['/usr/lib/esmond/esmond', '/usr/lib/esmond/lib/python2.7', '/usr/lib/esmond/lib/python2.7/site-packages', '/usr/lib/esmond/src/dlnetsnmp/lib', '/usr/lib/esmond', '/usr/lib/esmond/esmond_client', '/usr/lib64/python27.zip', '/usr/lib64/python2.7', '/usr/lib64/python2.7/plat-linux2', '/usr/lib64/python2.7/lib-tk', '/usr/lib64/python2.7/lib-old', '/usr/lib64/python2.7/lib-dynload', '/usr/lib64/python2.7/site-packages', '/usr/lib64/python2.7/site-packages/gtk-2.0', '/usr/lib/python2.7/site-packages']
[Fri Nov 16 09:26:23.468221 2018] [:error] [pid 103087] path= ['/usr/lib/esmond/esmond', '/usr/lib/esmond/lib/python2.7', '/usr/lib/esmond/lib/python2.7/site-packages', '/usr/lib/esmond/src/dlnetsnmp/lib', '/usr/lib/esmond', '/usr/lib/esmond/esmond_client', '/usr/lib64/python27.zip', '/usr/lib64/python2.7', '/usr/lib64/python2.7/plat-linux2', '/usr/lib64/python2.7/lib-tk', '/usr/lib64/python2.7/lib-old', '/usr/lib64/python2.7/lib-dynload', '/usr/lib64/python2.7/site-packages', '/usr/lib64/python2.7/site-packages/gtk-2.0', '/usr/lib/python2.7/site-packages']
[Fri Nov 16 09:26:23.474362 2018] [:error] [pid 103086] path= ['/usr/lib/esmond/esmond', '/usr/lib/esmond/lib/python2.7', '/usr/lib/esmond/lib/python2.7/site-packages', '/usr/lib/esmond/src/dlnetsnmp/lib', '/usr/lib/esmond', '/usr/lib/esmond/esmond_client', '/usr/lib64/python27.zip', '/usr/lib64/python2.7', '/usr/lib64/python2.7/plat-linux2', '/usr/lib64/python2.7/lib-tk', '/usr/lib64/python2.7/lib-old', '/usr/lib64/python2.7/lib-dynload', '/usr/lib64/python2.7/site-packages', '/usr/lib64/python2.7/site-packages/gtk-2.0', '/usr/lib/python2.7/site-packages']
[Fri Nov 16 09:26:25.108221 2018] [:error] [pid 103091] path= ['/usr/lib/esmond/esmond', '/usr/lib/esmond/lib/python2.7', '/usr/lib/esmond/lib/python2.7/site-packages', '/usr/lib/esmond/src/dlnetsnmp/lib', '/usr/lib/esmond', '/usr/lib/esmond/esmond_client', '/usr/lib64/python27.zip', '/usr/lib64/python2.7', '/usr/lib64/python2.7/plat-linux2', '/usr/lib64/python2.7/lib-tk', '/usr/lib64/python2.7/lib-old', '/usr/lib64/python2.7/lib-dynload', '/usr/lib64/python2.7/site-packages', '/usr/lib64/python2.7/site-packages/gtk-2.0', '/usr/lib/python2.7/site-packages']
[Fri Nov 16 09:26:25.512906 2018] [:error] [pid 103085] path= ['/usr/lib/esmond/esmond', '/usr/lib/esmond/lib/python2.7', '/usr/lib/esmond/lib/python2.7/site-packages', '/usr/lib/esmond/src/dlnetsnmp/lib', '/usr/lib/esmond', '/usr/lib/esmond/esmond_client', '/usr/lib64/python27.zip', '/usr/lib64/python2.7', '/usr/lib64/python2.7/plat-linux2', '/usr/lib64/python2.7/lib-tk', '/usr/lib64/python2.7/lib-old', '/usr/lib64/python2.7/lib-dynload', '/usr/lib64/python2.7/site-packages', '/usr/lib64/python2.7/site-packages/gtk-2.0', '/usr/lib/python2.7/site-packages']
[Fri Nov 16 09:26:27.455963 2018] [:error] [pid 103083] path= ['/usr/lib/esmond/esmond', '/usr/lib/esmond/lib/python2.7', '/usr/lib/esmond/lib/python2.7/site-packages', '/usr/lib/esmond/src/dlnetsnmp/lib', '/usr/lib/esmond', '/usr/lib/esmond/esmond_client', '/usr/lib64/python27.zip', '/usr/lib64/python2.7', '/usr/lib64/python2.7/plat-linux2', '/usr/lib64/python2.7/lib-tk', '/usr/lib64/python2.7/lib-old', '/usr/lib64/python2.7/lib-dynload', '/usr/lib64/python2.7/site-packages', '/usr/lib64/python2.7/site-packages/gtk-2.0', '/usr/lib/python2.7/site-packages']
[Fri Nov 16 09:26:27.539270 2018] [:error] [pid 103088] path= ['/usr/lib/esmond/esmond', '/usr/lib/esmond/lib/python2.7', '/usr/lib/esmond/lib/python2.7/site-packages', '/usr/lib/esmond/src/dlnetsnmp/lib', '/usr/lib/esmond', '/usr/lib/esmond/esmond_client', '/usr/lib64/python27.zip', '/usr/lib64/python2.7', '/usr/lib64/python2.7/plat-linux2', '/usr/lib64/python2.7/lib-tk', '/usr/lib64/python2.7/lib-old', '/usr/lib64/python2.7/lib-dynload', '/usr/lib64/python2.7/site-packages', '/usr/lib64/python2.7/site-packages/gtk-2.0', '/usr/lib/python2.7/site-packages']
[Fri Nov 16 09:26:32.947989 2018] [:error] [pid 103092] path= ['/usr/lib/esmond/esmond', '/usr/lib/esmond/lib/python2.7', '/usr/lib/esmond/lib/python2.7/site-packages', '/usr/lib/esmond/src/dlnetsnmp/lib', '/usr/lib/esmond', '/usr/lib/esmond/esmond_client', '/usr/lib64/python27.zip', '/usr/lib64/python2.7', '/usr/lib64/python2.7/plat-linux2', '/usr/lib64/python2.7/lib-tk', '/usr/lib64/python2.7/lib-old', '/usr/lib64/python2.7/lib-dynload', '/usr/lib64/python2.7/site-packages', '/usr/lib64/python2.7/site-packages/gtk-2.0', '/usr/lib/python2.7/site-packages']
[Fri Nov 16 09:26:35.489182 2018] [:error] [pid 103090] path= ['/usr/lib/esmond/esmond', '/usr/lib/esmond/lib/python2.7', '/usr/lib/esmond/lib/python2.7/site-packages', '/usr/lib/esmond/src/dlnetsnmp/lib', '/usr/lib/esmond', '/usr/lib/esmond/esmond_client', '/usr/lib64/python27.zip', '/usr/lib64/python2.7', '/usr/lib64/python2.7/plat-linux2', '/usr/lib64/python2.7/lib-tk', '/usr/lib64/python2.7/lib-old', '/usr/lib64/python2.7/lib-dynload', '/usr/lib64/python2.7/site-packages', '/usr/lib64/python2.7/site-packages/gtk-2.0', '/usr/lib/python2.7/site-packages']
[Fri Nov 16 09:26:36.272637 2018] [:error] [pid 103090] cassandra_db [INFO] Checking/creating column families
[Fri Nov 16 09:26:36.279580 2018] [:error] [pid 103090] cassandra_db [INFO] Schema check done
[Fri Nov 16 09:26:36.279650 2018] [:error] [pid 103090] cassandra_db [DEBUG] Opening ConnectionPool
[Fri Nov 16 09:26:36.409281 2018] [:error] [pid 103090] cassandra_db [INFO] Connected to ['localhost:9160']
[Fri Nov 16 09:26:36.498544 2018] [:error] [pid 103090] cassandra_db [INFO] Checking/creating column families
[Fri Nov 16 09:26:36.498583 2018] [:error] [pid 103090] cassandra_db [INFO] Checking/creating column families
[Fri Nov 16 09:26:36.540497 2018] [:error] [pid 103090] cassandra_db [INFO] Schema check done
[Fri Nov 16 09:26:36.540535 2018] [:error] [pid 103090] cassandra_db [INFO] Schema check done
[Fri Nov 16 09:26:36.540583 2018] [:error] [pid 103090] cassandra_db [DEBUG] Opening ConnectionPool
[Fri Nov 16 09:26:36.540600 2018] [:error] [pid 103090] cassandra_db [DEBUG] Opening ConnectionPool
[Fri Nov 16 09:26:36.649472 2018] [:error] [pid 103090] cassandra_db [INFO] Connected to ['localhost:9160']
[Fri Nov 16 09:26:36.649510 2018] [:error] [pid 103090] cassandra_db [INFO] Connected to ['localhost:9160']
[Fri Nov 16 09:26:36.861827 2018] [:error] [pid 103090] cassandra_db [DEBUG] Flush called
[Fri Nov 16 09:26:36.861871 2018] [:error] [pid 103090] cassandra_db [DEBUG] Flush called
[Sun Nov 18 03:10:02.210114 2018] [mpm_prefork:notice] [pid 103082] AH00171: Graceful restart requested, doing restart
[Sun Nov 18 03:10:03.059488 2018] [so:warn] [pid 103082] AH01574: module ssl_module is already loaded, skipping

```

**Comment by arlake228** [comment]
This indeed appears to be some type of bug with the LWP library being used by the call in question. I can easily recreate this error by trying to connect to an MA using https. I either get "write_failed" or I get "Inability to connect to HOSTNAME:443" in the cases I tried. I think the write failed may be related to the size of the request being written to the socket. Whatever the cause, this is the type of think LWP should be able to handle. As of 4.1, we have replaced LWP in most areas of the code for this reason along with poor IPv6 support. The fix here is for us to replace LWP with a call to a perfSONAR library call that handles the https connection properly. I have merged a change into 4.1.4 that I am able to confirm resolves this issue.

I built a new graphs package in nightly that can be installed as a temporary workaround with `yum install https://perfsonar-dev3.grnoc.iu.edu/nightly/el/7/x86_64/perfsonar/perfsonar-graphs-4.1.3-1.el7.643.15af3e66c8f5b0caff5c7ae9eec2a344a702b3b1.noarch.rpm`. That should make your errors go away.

 I am closing this issue for tracking purposes since I am able to test it and confirm it works. We can re-visit if others see different results.
